src/adapters/direction_api.js

```diff
@@ -74,9 +74,15 @@ export default class DirectionApi {
     try {
       response = await Ajax.get(directionsUrl, directionsParams, { timeout });
     } catch (e) {
-      return;
+      if (e >= 400 && e < 600) {
+        // Use the error codes 4xx and 5xx to display different error messages
+        response = { data: null, error: true, errorCode: e };
+      } else {
+        // Other errors are catched silently
+        return;
```

Comment:
Now that we have a `error` flag available for this use, this path should use it too to avoid data flow inconsistency.